src/compass/utils/geometry_utils.py

```diff
@@ -265,3 +265,163 @@ def get_unit_vector4component_of_interest(los_inc_angle, los_az_angle, comp='enu
+
+        vec_e, vec_n, vec_u = get_enu_vector_ecef(lon_deg, lat_deg)
+
+        llh_ref = np.array([np.deg2rad(lon_deg),
+                            np.deg2rad(lat_deg),
+                            hgt])
+
+        xyz_before = ellipsoid.lon_lat_to_xyz(llh_ref)
+        xyz_after_set = (xyz_before
+                         + vec_e * e_arr[index_arr]
+                         + vec_n * n_arr[index_arr]
+                         + vec_u * u_arr[index_arr])
+        llh_displaced = ellipsoid.xyz_to_lon_lat(xyz_after_set)
+
+        aztime_ref, slant_range_ref =\
+            isce3.geometry.geo2rdr(llh_ref,
+                                   ellipsoid,
+                                   orbit,
+                                   isce3.core.LUT2d(),
+                                   radargrid_ref.wavelength,
+                                   radargrid_ref.lookside,
+                                   threshold=threshold,
+                                   maxiter=maxiter)
+
+        aztime_displaced, slant_range_displaced =\
+            isce3.geometry.geo2rdr(llh_displaced,
+                                   ellipsoid,
+                                   orbit,
+                                   isce3.core.LUT2d(),
+                                   radargrid_ref.wavelength,
+                                   radargrid_ref.lookside,
+                                   threshold=threshold,
+                                   maxiter=maxiter)
+
+        rg_arr[index_arr] = slant_range_displaced - slant_range_ref
+        az_arr[index_arr] = aztime_displaced - aztime_ref
+
+    return rg_arr, az_arr
+
+
+def get_enu_vector_ecef(lon, lat, units='degrees'):
+    '''
+    Calculate the east, north, and up vectors in ECEF for lon / lat provided
+
+    Parameters
+    ----------
+    lon: np.ndarray
+        Longitude of the points to calculate ENU vectors
+    lat: np.ndarray
+        Latitude of the points to calculate ENU vectors
+    units: str
+        Units of the `lon` and `lat`.
+        Acceptable units are `radians` or `degrees`, (Default: degrees)
+
+    Returns
+    -------
+    vec_e: np.ndarray
+        unit vector of "east" direction in ECEF
+    vec_n: np.ndarray
+        unit vector of "north" direction in ECEF
+    vec_u: np.ndarray
+        unit vector of "up" direction in ECEF
+    '''
+    if units == 'degrees':
+        lon_rad = np.deg2rad(lon)
+        lat_rad = np.deg2rad(lat)
+    elif units == 'radians':
+        lon_rad = lon
+        lat_rad = lat
+    else:
+        raise ValueError(f'"{units}" was provided for `units`, '
+                         'which needs to be either `degrees` or `radians`')
+
+    vec_u = np.array([np.cos(lon_rad) * np.cos(lat_rad),
+                      np.sin(lon_rad) * np.cos(lat_rad),
+                      np.sin(lat_rad)])
+
+    vec_n = np.array([-np.cos(lon_rad) * np.sin(lat_rad),
+                      -np.sin(lon_rad) * np.sin(lat_rad),
+                      np.cos(lat_rad)])
+
```

Comment:
If you've got some easy reference for how you wrote these, that would be helpful here

Reply:
Good idea. In fact I did my own calculations, but I've added couple of links, which is essentially similar to what I did.